### ISSUE TITLE
Never ending sync.

### ISSUE THREAD

**Issue by gpoturlu** [ethereum/mist#3416]
Hello! Is there any way you can help me to get my ethers in my wallet in order to transfer and leave your application? I am tired of waiting for complete synchronisation, this will never end, you are always 100 - 200 blocks in front of me, i will never synchronise this way. My laptop is unusable, full with data i don't use, my wallet is still empty after more than 5 days. I am tired, is there any thing you can do to let ma walk way from you and your system? I see my coins in etherscan but there is nothing i can do there to leave. Thank you. Best regards! Gabriel. 

- [ ] I've asked for help in the [Mist Gitter](http://gitter.im/ethereum/mist) before filing this issue.

```
Version: `0.9.3`
OS & Version: osx
Node version: `geth 1.7.2` 
Number of blocks synchronized: 265 blocks left but never ending...
```

**Comment by gpoturlu** [comment]
Yeah, figures, no answer yet, why would you... in the mean time I keep staying 200 blocks behind, never ending sync...

**Comment by cmtonkinson** [comment]
@gpoturlu this isn't a corporate call center with a 24x7 SLA. An issue report that's a half hour old during the very early hours in parts of the world and dead of night elsewhere isn't an unreasonable offense. You're more likely to get help without the passive aggression.

**Comment by gpoturlu** [comment]
OK, clear! Looking forward for help then. Thanks for your advice!

**Comment by EarthlyChimp** [comment]
Hi, I didnt want to create a new post but I have a similar issue.

I have been sent Ethereum to my wallet.
I had Mist working previously before rebuilding my pc.
I have installed Mist latest version from the website, opened it up and clicked "Launch Application" and syncing last bit. 
I dropped my wallet file into Mist and it appears.
I then changed it to "Sync with light Client( Beta)". 
I then restarted Mist and waited. 
After many hours it keeps rotating between "Looking for peers" and 
If I skip final sync and choose Launch application my wallet always shows 0 Eth, but I know Eth has been sent.

Any advice or help would be greatly appreciated.

Thanks,

**Comment by highree** [comment]
@gpoturlu i finally got it working with osx high sierra. This is what i did: 

1. Backup our keystore (/Users/xxx/Library/Ethereum/keystore/*)
2. Delete the Ethereum Wallet in Appplications
3. Delete /Users/xxx/Library/Ethereum/ (xxx is your username)
4. Delete /Users/xxx/Library/Application Support/Ethereum Wallet
5. Reinstall the new Wallet (DONT START IT YET)
6. Find the file in /Users/xxx/Library/Application Support/Ethereum Wallet/syncmode
7. Open it with textedit replace "fast" with "light" and save the file 
8. Start Ethereum Wallet now
9. Wait 5-10 minutes to see if you get blocks. 

You will notice that you will see "Peers", before you could just see a 0. 

The total worktime of this is 4 minutes of your life. 
Donations are welcome, if u want pls donate for sick children.

**Comment by EarthlyChimp** [comment]
Im on Windows unfortunately. Any advice for windows 10?

**Comment by highree** [comment]
@EarthlyChimp its the same way, though actually i dont know the correct pathes in windumb 10.

**Comment by cmtonkinson** [comment]
I had a similar issue this week - Mist was around 200 blocks shy of current, and hung there for hours. I wound up closing Mist, and discovered that there were still swarmjs daemons running after. I rebooted the system, brought Mist back up and whatever the issue was went away.

**Note:** The first time I tried this, Mist wound up resyncing the entire chain. Second time worked. What changed? No idea.

**Comment by mashternator** [comment]
Hi Everyone. I am dealing with this too. However; I am willing to try the steps outlined by @highree but am wondering if this will affect any pending transfers.

I tried transferring some ETH to MEW yesterday and the transfer is still pending. This on top of the never ending sync suggests the transfer will likely not happen.  However' I also dont want to mess anything up. Any advice?

**Comment by highree** [comment]
hi @mashternator ,

if you do what i did, you will loose that transactions but not the ether. so you can do this. you need to reissue your transactions.

**Comment by mashternator** [comment]
@highree Thanks. Will update once I have tried it.

**Comment by Proggie** [comment]
I had to switch to the light client because I was running of space and now it doesn't sync. @highree how do you do step 6 & 7 if you've deleted the directory in step 4? It won't get recreated until you start the app again.

**Comment by ianaz** [comment]
Same problem here. Never ending

**Comment by ruki70** [comment]
Same Problem Sync about 40 h

**Comment by madnight** [comment]
If you are on linux try `while true; do timeout 6000 geth --syncmode "light"; done` and plus `geth removedb` to remove corrupted blockchains. It depends you can get luck and the geth + mist finds peers, syncs the lightchain db and starts the wallet after only (<20) retries, but sometimes you are unlucky you need (50<) retires, however that's much faster than a full sync from scratch that takes up a week (geth --fast cache 2048)

**Comment by gpoturlu** [comment]
Hello! Long time, no see... Do you think it is difficult to synchronise? Wait until you will have to send from one wallet to another... more than 10 minutes for send confirmation and i don't know yet how many hours until you can see the ETH in the other wallet. Even if transaction is confirmed, you can still see the same ETH in you current wallet, how about that!? I don't know, I think this system, network, environment should work faster, this way is not working for me, at least. And since i don't do free advertising and we are still free to speak what we think and experience, let me tell you about my LTC transactions: 2 minutes to confirmation and another 3 minutes to see the LTC in the other wallet. Why are they able to do that and ETH not? For me the only word to define the ETH experience is: "waiting". All the best from my side and happy waiting...

**Comment by ianaz** [comment]
What are you talking about? Always took me less than 30 seconds in the last
month for many different transactions

**Comment by mashternator** [comment]
Not sure if anyone tried @highree steps. They worked enough for me to be able to send all of the ETH out of my cursed Ethereum Wallet - which was excellent. However; I had some GNT in that same wallet because I wanted to see if Ethereum Wallet really could store GNT. Unfortunately now I cant locate that wallet file and since reinstalling Ethereum wallet. I fear I likely saved the wrong Keystore file since the wallet keys arent the same (I had a couple of wallets and likely only saved one of them - the empty one) Or . I do have the private Key for the wallet in question but the file is nowhere to be found. I can see that wallet on Etherscan and see that the GNT is still there. Is there any way to retrieve a previous version of an Ethereum Wallet file? Its not a huge deal, but if anyone has any advice on if this can be achieved, I would be very grateful. Sorry for the noob problems.

**Comment by woodydeck** [comment]
> Why are they able to do that and ETH not? For me the only word to define the ETH experience is: "waiting".

Geth + Mist is a terrible user experience. Ethereum is very early, and most developers are not designers. I think your passive aggressiveness is warranted here. Ethereum is very fast, just use MEW or MetaMask for dapps for now. Everyone is trying, but it is bugs galore.

**Comment by alexvandesande** [comment]
Duplicate of https://github.com/ethereum/mist/issues/3097 
Merging threads.

**Comment by allyouracid** [comment]
I 100% agree with woodydeck. It's a really terrible UX… I can handle a blockchain bloating my disk (that's why I chose to download it), but occupying that much of my computer's resources and basically rendering the network connection useless is just annoying.
Are there people out there who are able to actually sync the whole chain? Are there any reports of anyone successfully synchronizing with the network, having his or her Ether showing up in the wallet?

Right now, it's doing the "Downloading chain structure" part, which is especially annoying as it shows "xxx of yyy", while xxx increasing, which is cool as it shows progress, but: yyy is increasing as well???
Right now, I'm at 30.000.000 chain structure and I just don't know for how long xxx will hunt yyy without ever reaching it.

Can anybody confirm that it is (not only theoretically, actual experiences pls) possible to do a full sync?

**Comment by rafaelosirs** [comment]
If you are in a pinch and have to deploy a contract you don't need the full node sync, just follow this simple steps instead (i found in a post and help me a lot):

1. Install https://metamask.io/ and
2. Navigate to https://remix.ethereum.org/

3. Click on settings -> choose your compiler (e.g. 0.4.19+commit.c4cbbb05). Note that optimization is unchecked.
4. Login to metmask and choose your network (i would suggest first testing using ropsten test network and using the buy button to get ETH from a free faucet)
5. Go back to the compile tab,
Delete the contents of the example contract,
Paste in your contract (it should auto compile).
6. Click on the run tab
If you contract have input data you have tu put that data on double quoute (").
Click on create.
7. Metamask will open, choose your gas price and click Submit to sign your transaction
Open metmask again, click on the transaction hash.
8. Once published. in etherscan go to the contract address and Click on Contract Code tab on etherscan, click Verify and Publish.
9. Set the name of the contract to the exact name as in remix, Compiler to the same exact version as in remix and set optimization to "No" if you didn't click it on remix in step 3.
You should now have a published contract, with source code on etherscan.

**Comment by fenderrex** [comment]
i disables SMART on my hard drive years ago and i think it causing trouble i have my wallet hard linked to 3 drives but my chain synced only on one drive and is always crashing mist because i think my only large enough drive is dying on me. I back up to an old web server i rent over the year but I am ordering a new drive. disk drives suck

**Comment by fenderrex** [comment]
i start gith in cmd with geth.exe --rpc --maxpeers 35 --cache=2048 --syncmode=fast

**Comment by mrcope** [comment]
I was having the same issue as described.  I was able to fix it by doing a small subset of highree's steps.
1. Delete chaindata folder
2. Modify the Application Support/Ethereum Wallet/syncmode from "fast" to "light" (or alternately on the Develop menu, select "Sync with light client (beta)" which modifies the syncmode file for you.
3. Restart Ethereum Wallet.

I started it before going to bed and by the time I got up it had synched (~6 hours).
